include/oneapi/tbb/task_arena.h

```diff
@@ -113,6 +120,11 @@ namespace d1 {
 static constexpr unsigned num_priority_levels = 3;
 static constexpr int priority_stride = INT_MAX / (num_priority_levels + 1);
 
+#if __TBB_PREVIEW_PARALLEL_PHASE
+static constexpr int leave_policy_trait_offset = 1;
+static constexpr int leave_policy_trait_mask = ~((1 << leave_policy_trait_offset) - 1);
```

Comment:
What is the problem here? First bit is reserved for `core_type_support_flag` thus we take (for now) remaining more significant bits for leave policies.

Reply:
I think you should just add `leave_policy_support_flag` next to `core_type_support_flag` with the value of 2 (or 1 << 1, if you like it more).

Reply:
My idea here was to map values from enumeration class `leave_policy` directly into bits in the `my_version_and_traits` field and vice versa. User will pass values from enumeration class anyway to set default leave policy for the arena, so in order to not duplicate any non-automatic policies as flags in unnamed enum such way is used. If any new policies are going to be added- just single enum for `leave_policy` will need to be updated in `task_arena`.

@akukanov Do I understand it correctly that you are suggesting to add `fast_leave_policy_flag` and just check if it is set or not instead?

Reply:
> @akukanov Do I understand it correctly that you are suggesting to add fast_leave_policy_flag and just check if it is set or not instead?
@isaevil Yes, I believe it is the same functionally but easier to use than keeping the offset and the mask for a position in the traits. And also it would be consistent with `core_type_support_flag`

Reply:
@akukanov Added `fast_leave_policy_flag` instead of previous implementation with `leave_policy_trait_mask`.